python/lsst/obs/fiberspectrograph/spectrum.py

```diff
@@ -0,0 +1,198 @@
+# This file is part of obs_fiberspectrograph
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+__all__ = ("FiberSpectrum",)
+
+import numpy as np
+import astropy.io.fits
+import fitsio
+import astropy.units as u
+from ._instrument import FiberSpectrograph
+import lsst.afw.image as afwImage
+
+
+class VisitInfo:
+    def __init__(self, md):
+        self.exposureTime = md["EXPTIME"]
+
+    def getExposureTime(self):
+        return self.exposureTime
+
+
+class Info:
+    def __init__(self, md):
+        self.visitInfo = VisitInfo(md)
+
+    def getVisitInfo(self):
+        return self.visitInfo
+
+
+class FiberSpectrum:
+    def __init__(self, wavelength, flux, md=None):
+        self.wavelength = wavelength
+        self.flux = flux
+        self.md = md
+
+        self.info = Info(md)
```

Comment:
Is this needed to allow our Task code to think that it's really working with an `Exposure`?

Reply:
It still seems to run fine when removing `self.info = Info(md) `

Reply:
My question is why you are trying to emulate visit info.

Reply:
Replaced by `ObservationInfo` from astro_metadata_translator.

Reply:
Ok, but I still don't know what your requirements are. Do your pipelines use this information? Are they expecting ObservationInfo or VisitInfo? If you don't need C++ ObservationInfo is probably the right thing to do (just like not using PropertyList unless you need to pass it to C++).

Reply:
I think the goal is to make sure the spectra look sufficiently like an exposure that it can be handled by similar code for processing (as with the IsrTask subclass on this ticket).

Reply:
If you are relying on VisitInfo being available in that exact form then you have to use it like that -- you do have to think about whether you serialize the VisitInfo itself or make it a read-only view of the metadata and on read always regenerate from the metadata. With the metadata you can use a dict since PropertyList from an Exposure can behave like a dict.

Reply:
The answer is that I don't know.  I'm not sure that we know yet what the processing of this data looks like (at least I don't have a clear idea).  My hop is that having it match the existing data reduces the amount of fiber-spectrograph specific code.

Reply:
We will mostly want to know the central wavelength (and maybe width).  So stats on the spectrum -- but that may not help here